### PR TITLE
Robotics lift extended to third deck

### PR DESCRIPTION
**Pull request by DreamySkrell**
[2023-07-24--dreamseeker_10953_dRN1n4DANl.webm](https://github.com/Aurorastation/Aurora.3/assets/107256943/1915b2c5-15f1-4798-9817-1c25cf34ae5b)

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!wip

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/shuttles/shuttle_console_multi_lift.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (SleepyGemmy) [inline code/modules/shuttles/shuttle_console_multi_lift.dm:1]:
```suggestion
```

File: code/modules/shuttles/shuttle_console_multi_lift.dm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	req_access = null
+	ui_template = "shuttle_control_console_multi_lift.tmpl"
+
+/obj/machinery/computer/shuttle_control/multi/lift/wall
+	icon_state = "lift_wall"
+
+/obj/machinery/computer/shuttle_control/multi/lift/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	var/datum/shuttle/autodock/shuttle = SSshuttle.shuttles[shuttle_tag]
+	if (!istype(shuttle))
+		to_chat(user,"<span class='warning'>Unable to establish link with the shuttle.</span>")
```
Comment (SleepyGemmy) [inline code/modules/shuttles/shuttle_console_multi_lift.dm:17]:
```suggestion
	if(!istype(shuttle))
		to_chat(user, SPAN_WARNING("Unable to establish link with the shuttle."))
```

File: code/modules/shuttles/shuttle_console_multi_lift.dm [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		return
+
+	var/list/data = get_ui_data(shuttle)
+
+	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, ui_template, "[shuttle_tag] Control", 470, 300)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
```
Comment (SleepyGemmy) [inline code/modules/shuttles/shuttle_console_multi_lift.dm:27]:
```suggestion
		ui.set_auto_update(TRUE)
```
unless it's not supported.

File: code/modules/shuttles/shuttles_multi_lift.dm [hunk 4]
```diff
@@ -0,0 +1,52 @@
+
```
Comment (SleepyGemmy) [inline code/modules/shuttles/shuttles_multi_lift.dm:1]:
```suggestion
```

File: code/modules/shuttles/shuttles_multi_lift.dm [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+// multi lift, meaning it goes between stops
+// if it goes from first stop to the last one
+// it goes one by one sequentially through the ones in the middle
+// order of stops is same as defined in destination_tags
+/datum/shuttle/autodock/multi/lift/
```
Comment (SleepyGemmy) [inline code/modules/shuttles/shuttles_multi_lift.dm:6]:
```suggestion
/datum/shuttle/autodock/multi/lift
```

File: code/modules/shuttles/shuttles_multi_lift.dm [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	final_location = destinations_cache[destination_key]
+	var/next_tag = get_next_destination_tag()
+	next_location = SSshuttle.get_landmark(next_tag)
+	..(next_location.name, user)
+	launch(user)
+
+/datum/shuttle/autodock/multi/lift/arrived()
+	if(final_location == current_location)
+		final_location = null
+		// TODO: play sound lol
```
Comment (SleepyGemmy) [inline code/modules/shuttles/shuttles_multi_lift.dm:43]:
```suggestion
```
should be in the PR if its intended to be implemented.

File: code/modules/shuttles/shuttles_multi_lift.dm [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+	else
+		var/next_tag = get_next_destination_tag()
+		next_location = SSshuttle.get_landmark(next_tag)
+		set_destination(next_location.name, null)
+		launch(in_use)
+	..()
+
+// TODO: handle emergency stop
+
```
Comment (SleepyGemmy) [inline code/modules/shuttles/shuttles_multi_lift.dm:52]:
```suggestion

```
should be in the PR if its intended to be implemented.

File: html/changelogs/DreamySkrell-robotics-lift-1111.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Robotics lift extended to third deck."
```
Comment (SleepyGemmy) [inline html/changelogs/DreamySkrell-robotics-lift-1111.yml:41]:
```suggestion
  - maptweak: "Robotics lift extended to the third deck."
```

File: code/modules/shuttles/shuttles_multi_lift.dm (hunk 7 above)
Comment (DreamySkrell) [reply code/modules/shuttles/shuttles_multi_lift.dm:50]: forgot about this one lol - implemented now

Comment (SleepyGemmy) [reply code/modules/shuttles/shuttles_multi_lift.dm:50]: the comment is still there.

Comment (DreamySkrell) [reply code/modules/shuttles/shuttles_multi_lift.dm:50]: I forgot to remove it 💀

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**